locust/stats.py

```diff
@@ -812,9 +823,89 @@ def distribution_csv():
     ))]
     for s in chain(sort_stats(runners.locust_runner.request_stats), [runners.locust_runner.stats.total]):
         if s.num_requests:
-            rows.append(s.percentile(tpl='"%s",%i,%i,%i,%i,%i,%i,%i,%i,%i,%i,%i,%i'))
+            rows.append(s.percentile(tpl='"%s","%s",%i,%i,%i,%i,%i,%i,%i,%i,%i,%i,%i,%i'))
+        else:
+            rows.append('"%s","%s",0,"N/A","N/A","N/A","N/A","N/A","N/A","N/A","N/A","N/A","N/A","N/A"' % (s.method, s.name))
+
+    return "\n".join(rows)
+
+def stats_history_csv_header():
+    """Headers for the stats history CSV"""
+
+    return ','.join((
+        '"Type"',
+        '"Name"',
+        '"Timestamp"',
+        '"# requests"',
+        '"# failures"',
+        '"Requests/s"',
+        '"Requests Failed/s',
+        '"Median response time"',
+        '"Average response time"',
+        '"Min response time"',
+        '"Max response time"',
+        '"Average Content Size"',
+        '"50%"',
+        '"66%"',
+        '"75%"',
+        '"80%"',
+        '"90%"',
+        '"95%"',
+        '"98%"',
+        '"99%"',
+        '"99.9%"',
+        '"99.99%"',
+        '"99.999"',
+        '"100%"'
+    )) + '\n'
+
+def stats_history_csv(stats_history_enabled=False):
+    """Returns the Aggregated stats entry every interval"""
+    from . import runners
+
+    rows = []
+    timestamp = int(time.time())
+    PERCENTILES_TO_REPORT = [
+        0.50,
+        0.66,
+        0.75,
+        0.80,
+        0.90,
+        0.95,
+        0.98,
+        0.99,
+        0.999,
+        0.9999,
+        0.99999,
+        1.0
+    ]
+
+    include_stats_entries_per_iteration = []
+    if stats_history_enabled:
+        include_stats_entries_per_iteration = sort_stats(runners.locust_runner.request_stats)
+
+    for s in chain(include_stats_entries_per_iteration, [runners.locust_runner.stats.total]):
+        if s.num_requests:
+            percentile_str = ','.join([
+                str(int(s.get_response_time_percentile(x) or 0)) for x in PERCENTILES_TO_REPORT])
```

Comment:
since we don't report float values for response times (its 640ms not 640.25 etc.) we are converting to int and then string for writing it to the csv (since we create all percentiles as 1 big string and pass it to rows(as rows.append) on line 907 (instead of each percentile value for each column)